File: examples/python/s3_create_bucket.py
"""Crea un bucket S3 nuevo en Floci.

Uso: python3 s3_create_bucket.py [nombre-del-bucket]
"""
import sys
import time

import boto3
from botocore.exceptions import ClientError

s3 = boto3.client(
    "s3",
    endpoint_url="http://localhost:4566",
    aws_access_key_id="test",
    aws_secret_access_key="test",
    region_name="us-east-1",
)


def main():
    bucket_name = sys.argv[1] if len(sys.argv) > 1 else f"mi-bucket-{int(time.time())}"
    try:
        s3.create_bucket(Bucket=bucket_name)
        print("Bucket creado:", bucket_name)
    except ClientError as error:
        if error.response["Error"]["Code"] == "BucketAlreadyOwnedByYou":
            print(f'El bucket "{bucket_name}" ya existe. Elige otro nombre o elimínalo primero.')
        else:
            raise


if __name__ == "__main__":
    main()

File: examples/node/s3-create-bucket.js
// Crea un bucket S3 nuevo en Floci.
// Uso: node s3-create-bucket.js [nombre-del-bucket]
const AWS = require('aws-sdk');

const s3 = new AWS.S3({
  endpoint: 'http://localhost:4566',
  region: 'us-east-1',
  credentials: { accessKeyId: 'test', secretAccessKey: 'test' },
  s3ForcePathStyle: true,
});

const bucketName = process.argv[2] || `mi-bucket-${Date.now()}`;

async function main() {
  await s3.createBucket({ Bucket: bucketName }).promise();
  console.log('Bucket creado:', bucketName);
}

main().catch((error) => {
  if (error.code === 'BucketAlreadyOwnedByYou') {
    console.error(`El bucket "${bucketName}" ya existe. Elige otro nombre o elimínalo primero.`);
  } else {
    console.error('Error creando el bucket:', error.message);
  }
  process.exit(1);
});
